polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -1478,17 +1474,28 @@ ValueCategory MLIRScanner::VisitCastExpr(CastExpr *E) {
     return Visit(E->getSubExpr());
 
   case clang::CastKind::CK_AddressSpaceConversion: {
-    auto Scalar = Visit(E->getSubExpr());
-    // JLE_QUEL::TODO (II-201)
-    // assert(scalar.isReference);
-    auto PostTy = ReturnVal.getType().cast<MemRefType>().getElementType();
-    return ValueCategory(castToMemSpaceOfType(Scalar.val, PostTy),
-                         Scalar.isReference);
+    ValueCategory Scalar = Visit(E->getSubExpr());
+    QualType DestTy = E->getType();
+    unsigned AS = Glob.getCGM().getContext().getTargetAddressSpace(
+        DestTy->isPointerType() ? DestTy->getPointeeType().getAddressSpace()
+                                : DestTy.getAddressSpace());
+#ifdef DEBUG
```

Comment:
If reviewers prefer we can remove the assertion code.

Reply:
I prefer to remove the assertion code, as the original code was added by us.